streams/jvm/src/main/scala/zio/stream/platform.scala

```diff
@@ -533,21 +542,30 @@ trait ZStreamPlatformSpecificConstructors {
      * Write the entire Chuck[Byte] to the socket channel.
      * The caller of this function is NOT responsible for closing the `AsynchronousSocketChannel`.
      *
-     * The sink will yield the count of bytes written.
+     * The sink will yield true if stream was completely written.
      */
-    def write: Sink[Throwable, Byte, Nothing, Int] =
-      ZSink.foldLeftChunksM(0) { case (nbBytesWritten, c) =>
-        IO.effectAsync[Throwable, Int] { callback =>
-          socket.write(
-            ByteBuffer.wrap(c.toArray),
-            null,
-            new CompletionHandler[Integer, Void] {
-              override def completed(result: Integer, attachment: Void): Unit =
-                callback(ZIO.succeed(nbBytesWritten + result.toInt))
-
-              override def failed(error: Throwable, attachment: Void): Unit = callback(ZIO.fail(error))
-            }
-          )
+    def write: Sink[Throwable, Byte, Byte, Boolean] =
```

Comment:
We can keep this returning the number of written bytes by keeping a var with the count on the CompletionHandler and incrementing it after every write.

What do you think?

Reply:
I had thought the byte count was not useful for network connections since they are not a fixed interaction (like a file).
The boolean does provide information that is not otherwise available and could have value for a caller.
However, avoiding unnecessary breakage of existing code is a concern.
I will adjust the code to restore the original functionality